### PR TITLE
Test on MacOS PHP 5.6

### PR DESCRIPTION
**Pull request by GrahamCampbell**
On actions, and remove the legacy CI.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -38,6 +38,11 @@ jobs:
             php-version: '5.6'
             SKIP_LINT_TEST_CASES: 1
 
+          - operating-system: 'macos-latest'
+            php-version: '5.6'
```
Comment (keradus) [inline .github/workflows/ci.yaml:42]: why?

Comment (GrahamCampbell) [reply .github/workflows/ci.yaml:42]: So we know it actually works on macos with php 5.6, just like we do for Windows.

When we drop PHP < 7.1, we can change both builds to 7.1.

Comment (keradus) [reply .github/workflows/ci.yaml:42]: i believe we do not need to check OS-specific behaviour on many PHP envs. one is enough

Comment (GrahamCampbell) [reply .github/workflows/ci.yaml:42]: So we should replace the windows 5.6 and 7.3 builds by a single 7.4 build?

Comment (keradus) [reply .github/workflows/ci.yaml:42]: Already in earlier pr

**Pull request closed without merging**